### ISSUE TITLE
Problem after increasing cosmological parameters

### ISSUE THREAD

**Issue by Don19** [brinckmann/montepython_public#86]
Hi! 

I am new to using the Monte Python. My project needs me to add cosmological parameters for calculation. But when I perform MCMC operation, I find that there are still six cosmological parameters in the .param file in the output folder.  It does not consider the parameters I added in the baseXXX.param. So what documents do I need to modify ?

Any idea will be very appreciated, thanks a lot!

**Comment by brinckmann** [comment]
Hi,

Can you please make a careful description of what you did?

Best,
Thejs

**Comment by Don19** [comment]
> Hi,
> 
> Can you please make a careful description of what you did?
> 
> Best,
> Thejs

Thanks for your reply.

I  used the idea called parameter splitting, which means I copied another set of parameters (To be precise,struct  background ba ). So I can test the consistency of some models.

I created another struct background ba and assigned values in the input file. Except for the parameters to be verified (such as w0_fld and w0_fld_growth , the last one is the new parameter I created), the other parameters are assigned the same value. I calculated the background model twice in the main function using the two sets of background parameters.  Then I using the new parameters （background ba2）in the perturbation model, and other calculation processes are unchanged. Finally, I modified the classy.pyx to make the calculation process the same as the main function.

 When I use the Monte Python, an error is displayed as follows:
Error in Class: perturb_init(L:196) :condition (w_fld_ini >= 0.) is true; The fluid is meant to be negligible at early time, and unimportant for defining the initial conditions of other species. You are using parameters for which this assumption may break down, since at early times you have w_fld(a--->0) = 6.953329e-310 >= 0

And I found that there is no w0_fld and w0_fld_gro  in the .paramnames file in the output folder although I had defined them  in the classy.pyx file and in the baseXXX.param. Apparently, I may have made an obvious mistake, but I really can't find out where I made the mistake. Can you please give me some suggestions?

**Comment by brinckmann** [comment]
Hi,

Can you please copy your parameter file?

Also, the error looks like maybe you need to add some suitable boundary conditions? I guess it really depends on what your new parameter does. Also keep in mind when running with w0_fld you need to set Omega_Lambda to zero in order for the parameter to be used, i.e. adding in the param file:
data.cosmo_arguments['Omega_Lambda'] = 0

Best,
Thejs

**Comment by Don19** [comment]
> Hi,
> 
> Can you please copy your parameter file?
> 
> Also, the error looks like maybe you need to add some suitable boundary conditions? I guess it really depends on what your new parameter does. Also keep in mind when running with w0_fld you need to set Omega_Lambda to zero in order for the parameter to be used, i.e. adding in the param file:
> data.cosmo_arguments['Omega_Lambda'] = 0
> 
> Best,
> Thejs

Thank you, Thejs.

 Indeed I did not set Omega_Lambda to zero.  However, I find that even if I added this line and used the Class without modification (except definng  w0_fld in the classy.pyx file), the Monte Python still does not run w0_fld whether I set this parameter to cosmo or derived . I guess maybe this is the same problem as my problem.

I made a copy of the files I modified except the main function and the header file. All the parts I modified were added with a "Donz" note. Thank you a lot for taking the time to help me solve the problem.

Best,
Don

[20191128.zip](https://github.com/brinckmann/montepython_public/files/3899777/20191128.zip)
